internal/providers/terraform/google/storage_bucket.go

```diff
@@ -1,215 +1,22 @@
-	if d.Get("storage_class").Exists() {
-		storageClass = d.Get("storage_class").String()
-	}
-
-	storageClassResourceGroupMap := map[string]string{
-		"STANDARD":       "RegionalOps",
-		"REGIONAL":       "RegionalOps",
-		"MULTI_REGIONAL": "MultiRegionalOps",
-		"NEARLINE":       "NearlineOps",
-		"COLDLINE":       "ColdlineOps",
-		"ARCHIVE":        "ArchiveOps",
-	}
-
-	return []*schema.CostComponent{
-		{
-			Name:            "Object adds, bucket/object list (class A)",
-			Unit:            "10k operations",
-			UnitMultiplier:  decimal.NewFromInt(10000),
-			MonthlyQuantity: classAQuantity,
-			ProductFilter: &schema.ProductFilter{
-				VendorName: strPtr("gcp"),
-				Service:    strPtr("Cloud Storage"),
-				AttributeFilters: []*schema.AttributeFilter{
-					{Key: "resourceGroup", Value: strPtr(storageClassResourceGroupMap[storageClass])},
-					{Key: "description", ValueRegex: strPtr("/Class A/")},
-				},
-			},
-			PriceFilter: &schema.PriceFilter{
-				EndUsageAmount: strPtr(""),
-			},
-		},
-		{
-			Name:            "Object gets, retrieve bucket/object metadata (class B)",
-			Unit:            "10k operations",
-			UnitMultiplier:  decimal.NewFromInt(10000),
-			MonthlyQuantity: classBQuantity,
-			ProductFilter: &schema.ProductFilter{
-				VendorName: strPtr("gcp"),
-				Service:    strPtr("Cloud Storage"),
-				AttributeFilters: []*schema.AttributeFilter{
-					{Key: "resourceGroup", Value: strPtr(storageClassResourceGroupMap[storageClass])},
-					{Key: "description", ValueRegex: strPtr("/Class B/")},
-				},
-			},
-			PriceFilter: &schema.PriceFilter{
-				EndUsageAmount: strPtr(""),
-			},
-		},
-	}
-}
-
-func dataRetrieval(d *schema.ResourceData, u *schema.UsageData) *schema.CostComponent {
-	var quantity *decimal.Decimal
-	if u != nil && u.Get("monthly_data_retrieval_gb").Exists() {
-		quantity = decimalPtr(decimal.NewFromInt(u.Get("monthly_data_retrieval_gb").Int()))
-	}
-
-	storageClass := "STANDARD"
-	if d.Get("storage_class").Exists() {
-		storageClass = d.Get("storage_class").String()
-	}
-
-	storageClassResourceGroupMap := map[string]string{
-		"NEARLINE": "NearlineOps",
-		"COLDLINE": "ColdlineOps",
-		"ARCHIVE":  "ArchiveOps",
-	}
-	resourceGroup := storageClassResourceGroupMap[storageClass]
-	// Skipping standard, regional and multi-regional since they are free
-	if resourceGroup == "" {
-		return nil
-	}
-
-	return &schema.CostComponent{
-		Name:            "Data retrieval",
-		Unit:            "GB",
-		UnitMultiplier:  decimal.NewFromInt(1),
-		MonthlyQuantity: quantity,
-		ProductFilter: &schema.ProductFilter{
-			VendorName: strPtr("gcp"),
-			Service:    strPtr("Cloud Storage"),
-			AttributeFilters: []*schema.AttributeFilter{
-				{Key: "resourceGroup", Value: strPtr(resourceGroup)},
-				{Key: "description", ValueRegex: strPtr("/Retrieval/")},
-			},
-		},
+	r := &google.StorageBucket{Address: d.Address, Region: d.Get("region").String(), Location: d.Get("location").String()}
+	if !d.IsEmpty("storage_class") {
```

Comment:
As above this can be moved to the line above.

Reply:
fixed